Clarify requirements for render-blocking (#39809)
mdn/content: files/en-us/web/api/htmllinkelement/blocking/index.md

A string. Must be a space-separated list of blocking tokens listed below indicating the operations that are to be blocked:

- `render`

> - : The rendering of content on the screen is blocked.
> 
>   > [!NOTE]
>   > In order to block rendering, `link` elements with `blocking="render"` must be in the `head` of the document.
>   > `link` elements with `rel="stylesheet"` that are in the `head` are implicitly render-blocking, unless they are added to the document via script. `link` elements with `rel="stylesheet"` added via script need an explicit `blocking="render"` in order to block rendering.

Reviewer: @noamr would this rephrasing help?

@eeeps the note is [breaking the formatting](https://pr39809.review.mdn.allizom.net/en-US/docs/Web/API/HTMLLinkElement/blocking#value) of the definition list just before it; I've added an extra blank line - hopefully that should fix it.

```suggestion
- : The rendering of content on the screen is blocked.
- 
-  > [!NOTE]
    > To block rendering, you must include the `link` element with `blocking="render"` in the document's `<head>`.
    > By default, a `link` element with `rel="stylesheet"` in the `<head>` blocks rendering when the browser discovers it during parsing.
    > If a stylesheet is added dynamically via script, you must also add it to the `<head>` and include `blocking="render"` for it to block rendering.
```

>   > `link` elements with `rel="stylesheet"` that are in the `head` are implicitly render-blocking, unless they are added to the document via script. `link` elements with `rel="stylesheet"` added via script need an explicit `blocking="render"` in order to block rendering.

Reviewer 2: ```suggestion
  > `link` elements with `rel="stylesheet"` that are in the `head` are implicitly render-blocking, if discovered at parse time. Such elements added via script need an explicit `blocking="render"` in order to block rendering.
```
